state/protocol/badger/state.go

```diff
@@ -164,43 +149,43 @@ func Bootstrap(
 		}
 
 		// 3) initialize the current protocol state height/view pointers
-		err = transaction.WithTx(state.bootstrapStatePointers(root))(tx)
+		err = transaction.WithTx(bootstrapStatePointers(root))(tx)
 		if err != nil {
 			return fmt.Errorf("could not bootstrap height/view pointers: %w", err)
 		}
 
 		// 4) initialize values related to the epoch logic
-		err = state.bootstrapEpoch(root.Epochs(), !config.SkipNetworkAddressValidation)(tx)
+		err = bootstrapEpoch(setups, commits, root.Epochs(), !config.SkipNetworkAddressValidation)(tx)
 		if err != nil {
 			return fmt.Errorf("could not bootstrap epoch values: %w", err)
 		}
 
 		// 5) initialize spork params
-		err = transaction.WithTx(state.bootstrapSporkInfo(root))(tx)
+		err = transaction.WithTx(bootstrapSporkInfo(root))(tx)
```

Comment:
I feel we are mixing API patterns here adding unnecessary complexity to the code. I would suggest that all service methods just operate on `tx`. They all have the ability to utilize callbacks, even if they are internally not using it. 

I have tried to remove the `transaction.WithTx` in a variety of places, which I feel declutters the code and improves readability.

Reply:
Tried to consolidate usages and left it only in places where it's used a lot: https://github.com/onflow/flow-go/pull/5113/commits/3c2d1589bb6c786b8baedcfa4b54eaa3186fcae3